enarx-keep/src/backend/kvm/vm/cpu.rs

```diff
@@ -107,18 +110,60 @@ impl Thread for Cpu {
         match self.fd.run()? {
             VcpuExit::IoOut(port, _) => match port {
                 SYSCALL_TRIGGER_PORT => {
-                    let keep = self.keep.write().unwrap();
-                    let sallyport = {
-                        let page = keep
-                            .address_space
+                    let mut keep = self.keep.write().unwrap();
+                    let mut sallyport = {
+                        let page = keep.regions[0]
                             .prefix_mut()
                             .shared_pages
                             .get_mut(self.id)
                             .unwrap();
 
                         unsafe { &mut *(page as *mut Page as *mut sallyport::Block) }
                     };
-                    Ok(Command::SysCall(sallyport))
+                    let syscall_nr: i64 = unsafe { sallyport.msg.req.num.into() };
+
+                    match syscall_nr {
```

Comment:
I don't understand what you mean? `Command::from(i64)` does not make any sense to me.

Reply:
At the time I was just trying to think of ways to factor out this match statement because over time it's probably just going to get bigger and bigger which will turn this function into quite a large one.